src/zulip.rs

```diff
@@ -262,29 +270,58 @@ async fn handle_command<'a>(
             StreamCommand::Read => post_waiter(&ctx, message_data, WaitingMessage::start_reading())
                 .await
                 .map_err(|e| format_err!("Failed to await at this time: {e:?}")),
-            StreamCommand::PingGoals(args) => ping_goals_cmd(ctx, gh_id, &args).await,
+            StreamCommand::PingGoals(args) => ping_goals_cmd(ctx, gh_id, message_data, &args).await,
             StreamCommand::DocsUpdate => trigger_docs_update(message_data, &ctx.zulip),
         }
     }
 }
 
 async fn ping_goals_cmd(
-    ctx: &Context,
+    ctx: Arc<Context>,
     gh_id: u64,
+    message: &Message,
     args: &PingGoalsArgs,
 ) -> anyhow::Result<Option<String>> {
     if project_goals::check_project_goal_acl(&ctx.team, gh_id).await? {
-        ping_project_goals_owners(
-            &ctx.github,
-            &ctx.zulip,
-            &ctx.team,
-            false,
-            args.threshold as i64,
-            &format!("on {}", args.next_update),
-        )
-        .await
-        .map_err(|e| format_err!("Failed to await at this time: {e:?}"))?;
-        Ok(None)
+        let args = args.clone();
+        let message = message.clone();
+        tokio::spawn(async move {
+            let res = ping_project_goals_owners(
+                &ctx.github,
+                &ctx.zulip,
+                &ctx.team,
+                false,
+                args.threshold as i64,
+                &format!("on {}", args.next_update),
+            )
+            .await;
+
+            let status = match res {
+                Ok(_res) => "OK".to_string(),
+                Err(err) => {
+                    tracing::error!("ping_project_goals_owners: {err:?}");
+                    format!("ERROR\n\n```\n{err:#?}\n```\n")
+                }
+            };
+
+            let res = MessageApiRequest {
+                recipient: Recipient::Private {
+                    id: message.sender_id,
+                    email: &message.sender_email,
+                },
```

Comment:
This is not actually right, the command can be sent from both private message and stream. I need to change that.

Reply:
Good catch, I forgot that, I find it kind of weird that it can be sent from both, but it is indeed supported.

Reply:
Fixed.